corehq/apps/case_importer/do_import.py

```diff
@@ -453,68 +449,69 @@ def _populate_updated_fields(config, row):
     return fields_to_update
 
 
-def _check_owner(owner, domain, user_id=None, locations=ALL_LOCATIONS):
-    owner_is_user = isinstance(owner, CouchUser) and owner.is_member_of(domain)
-    owner_is_casesharing_group = isinstance(owner, Group) and owner.case_sharing and owner.is_member_of(domain)
-    if not (owner_is_user or owner_is_casesharing_group or _is_valid_location_owner(owner, domain)):
-        raise exceptions.InvalidOwnerId('owner_id')
-    if not _is_owner_location_accessible_to_user(owner, domain, user_id, locations):
-        raise exceptions.InvalidLocation('owner_id')
-    return True
-
-
-def _is_valid_location_owner(owner, domain):
-    return (
-        isinstance(owner, SQLLocation) and
-        owner.domain == domain and
-        owner.location_type.shares_cases
-    )
-
-
-def _is_owner_location_accessible_to_user(owner, domain, user_id, locations_accessible_to_user):
-    return (
-        owner._id == user_id or
-        locations_accessible_to_user == ALL_LOCATIONS or
-        owner._id in locations_accessible_to_user or
-        (
-            hasattr(owner, 'get_location_id')  # is a user, not a location
-            and owner.get_location_id(domain) in locations_accessible_to_user
+class _OwnerAccessor(object):
```

Comment:
I recommend [?w=1](https://github.com/dimagi/commcare-hq/pull/25345/commits/799a457d7a693b78dc66c32d1b8dada86cbb3789?w=1) for viewing this change - it's mostly indentation